src/LuaTemplater/LuaTemplater.cpp

```diff
@@ -23,196 +23,190 @@ using namespace std;
-      "            if not pos then\n"
-      "                appender(builder, \"End tag ('}}') missing\")\n"
-      "                break\n"
-      "            end\n"
-      "            run_block(builder, tmpl:sub(b, pos+1))\n"
-      "            -- Skip back the }} (pos points to the start of }}).\n"
-      "            pos = pos+2\n"
-      "        elseif bracepercentpos then\n"
-      "            b = bracepercentpos\n"
-      "            appender(builder, tmpl:sub(pos, b-1))\n"
-      "            -- Find the end of the block.\n"
-      "            pos = tmpl:find(\"%%}\", b)\n"
-      "            if not pos then\n"
-      "                appender(builder, \"End tag ('%}') missing\")\n"
-      "                break\n"
-      "            end\n"
-      "            run_block(builder, tmpl:sub(b, pos+1))\n"
-      "            -- Skip back the }} (pos points to the start of }}).\n"
-      "            pos = pos+2\n"
-      "        else\n"
-      "            break\n"
-      "        end\n"
-      "    end\n"
-      "    -- Add any text after the last block. Or all of it if there\n"
-      "    -- are no blocks.\n"
-      "    if pos then\n"
-      "        appender(builder, tmpl:sub(pos, #tmpl))\n"
-      "    end\n"
-      "\n"
-      "    builder[#builder+1] = \"return table.concat(_ret)\"\n"
-      "    local tc = table.concat(builder, \"\\n\")\n"
-      "    -- print('tc', tc)\n"
-      "    local func, err = load(tc)\n"
-      "    if not func then\n"
-      "        print('Error')\n"
-      "        print('Incoming template:\\n', tmpl)\n"
-      "        print('intemediate form (this is just pure Lua code, you can "
-      "run this in lua):\\n', tc)\n"
-      "        return err\n"
-      "    end\n"
-      "    status, res = pcall(\n"
-      "      function()\n"
-      "        return func()\n"
-      "      end\n"
-      "    )\n"
-      "    if not status then\n"
-      "      print('Error')\n"
-      "      print('Incoming template:\\n', tmpl)\n"
-      "      print('intemediate form (this is just pure Lua code, you can run "
-      "this in lua, see what happens):')\n"
-      "      i = 1\n"
-      "      for line in string.gmatch(tc, '[^\\n]+') do\n"
-      "        print(i .. ':', line)\n"
-      "        i = i + 1\n"
-      "      end\n"
-      "      print('err', res)\n"
-      "      error( res )\n"
-      "    else\n"
-      "      return res\n"
-      "    end\n"
-      "end\n"
-      "\n"
-      "function M.compile(tmpl)\n"
-      "  -- status = true\n"
-      "  last_compiled = ''\n"
-      "  compiled = tmpl\n"
-      "  while compiled ~= last_compiled do\n"
-      "    last_compiled = compiled\n"
-      "    compiled = M._compile(compiled)\n"
-      "  end\n"
-      "  return compiled\n"
-      "end\n"
-      "\n"
-      "return M\n"
-      "\n"
-      "";
+  const char * templaterSource =  
+  "-- adapted from https://john.nachtimwald.com/2014/08/06/using-lua-as-a-templating-engine/\n" 
```

Comment:
observaiton: if we're using c++11 (are we?), this could probably be a raw-string, ie the `R"(...)"` syntax